File: Toy_Language_Interpreter/Repository/Repository.java
package Repository;

import ADTs.MyList;
import Exceptions.EmptyContainerException;
import Model.ProgramState;

import java.io.*;

public class Repository implements IRepo{
    private MyList<ProgramState> states;
    private String logFilePath;

    public Repository(String filePath) {
        this.states = new MyList<>();
        this.logFilePath = filePath;
    }

    public ProgramState getCrtPrg() throws EmptyContainerException {
        return this.states.remove();
    }


    public boolean isEmpty() {
        return states.isEmpty();
    }

    public void add(ProgramState state) {
        this.states.addFirst(state);
    }

    public void logPrgStateExec(ProgramState state) throws IOException {
        FileWriter fileName = new FileWriter(this.logFilePath, true);
        BufferedWriter buff = new BufferedWriter(fileName);
        PrintWriter logFile = new PrintWriter(buff);
        logFile.println(state.toString());
        logFile.close();
    }
}

File: Toy_Language_Interpreter_C/Repository/Repository.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class Repository : IRepo
    {
        private MyList<ProgramState> states;
        public MyList<ProgramState> States
        {
            get
            {
                return this.states;
            }
        }

        private String logFilePath;

        public Repository(String filePath)
        {
            this.states = new MyList<ProgramState>();
            this.logFilePath = filePath;
        }

        public ProgramState GetCrtPrg()
        {
            return this.states.Remove();
        }

        public bool IsEmpty()
        {
            return this.states.IsEmpty();
        }

        public void Add(ProgramState state)
        {
            this.states.AddFirst(state);
        }


        public void LogPrgStateExec(ProgramState state)
        {
            String fullPath = "D:\\facultate\\anul II\\metode avansate de programare\\laborator\\toy_language_interpreter_C#\\a7\\a7\\Text Files\\" + this.logFilePath;
            FileStream writeStream = new FileStream(fullPath, FileMode.Append);
            TextWriter writer = new StreamWriter(writeStream);
            writer.WriteLine(state.ToString());
            writer.Close();
        }
    }
}
